src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -85,15 +87,18 @@ internal SafeSslContextHandle(IntPtr handle, bool ownsHandle)
         {
         }
 
-        public override bool IsInvalid
-        {
-            get { return handle == IntPtr.Zero; }
-        }
+        public GCHandle AlpnHandle { get => _alpnHandle; set => _alpnHandle = value; }
```

Comment:
Could just be:
```C#
public GCHandle AlpnHandle { get; set; }
```
and then you wouldn't need to explicitly define the field.

Reply:
its a struct right? what happens if you free the struct from the get? isn't a copy that now isn't set to free?

I have been caught out on GCHandle before with Readonly, in that the copy updates the "Free" but the original doesn't thus causing a double free and an exception.

Reply:
I don't understand the questions.  The code:
```C#
public GCHandle AlpnHandle { get; set; }
```
is functionally identical to:
```C#
private GCHandle _alpnHandle;
public GCHandle AlpnHandle { get => _alpnHandle; set => _alpnHandle = value; }
```
The only difference is what the name of the field is, as the compiler will generate that field for you in the first case.

Reply:
``` csharp
if(_alpnHandle.IsAllocated)
 {
              _alpnHandle.Free();
 }
```

Reply:
If I do that, and then call IsAllocated straight after and its a property, wont isallocated still be true?

Reply:
Ah, you're concerned about a subsequent call to ReleaseHandle seeing AlpnHandle.IsAllocated as true and trying to free it again?  Yes, that could happen, so if that's the concern, then yeah, you could stick with what you have.  Though ReleaseHandle should not be called multiple times.

Reply:
Your call, I don't mind either way

Reply:
What you have is fine.